packages/myst-frontmatter/src/frontmatter/types.ts

```diff
@@ -72,16 +72,13 @@ export type Thebe = {
   local?: boolean | JupyterLocalOptions;
 };
 
-export enum BinderProviders {
-  git = 'git',
-  github = 'github',
-  gitlab = 'gitlab',
-}
+export type WellKnownRepoProviders = 'github' | 'gitlab' | 'git' | 'gist';
+export type BinderProviders = WellKnownRepoProviders | string;
 
 export type BinderHubOptions = {
   url?: string;
-  ref?: string; // org-name/repo-name
-  repo?: string; // valid git refs only?
+  ref?: string; // org-name/repo-name for WellKnownRepoProviders, url for 'meca', otherwise any string
```

Comment:
These comments don't quite seem to match the code where `ref` is validated as any string, regardless of provider, and `repo` is validated as either a github string or arbitrary string, depending.

However, the actual code functionality seems consistent with the previous behaviour (i.e. `ref` is still always validated as a simple string, `repo` has more complex logic) so I will go ahead and merge. Not going to change this comment in case there is something I am misunderstanding.

Reply:
yes that comment is wrong, I'll remove on the next pass